src/test/java/book_store/controller/BookControllerTest.java

```diff
@@ -0,0 +1,233 @@
+    void createBook_CreatesNewBook_ExpectedSuccess() throws Exception {
+        CreateBookRequestDto requestDto = new CreateBookRequestDto()
+                .setAuthor("John Doe")
+                .setTitle("Book Title")
+                .setDescription("Book Description")
+                .setIsbn("9460306332081")
+                .setPrice(BigDecimal.valueOf(10.99));
+
+        BookDto expected = new BookDto()
+                .setAuthor(requestDto.getAuthor())
+                .setTitle(requestDto.getTitle())
+                .setDescription(requestDto.getDescription())
+                .setIsbn(requestDto.getIsbn())
+                .setPrice(requestDto.getPrice());
+
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+
+        MvcResult result = mockMvc.perform(
+                post("/books")
+                    .content(jsonRequest)
+                    .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+        BookDto actual = objectMapper.readValue(result.getResponse().getContentAsString(), BookDto.class);
+        Assertions.assertNotNull(actual);
+        Assertions.assertNotNull(actual.getId());
+        Assertions.assertEquals(expected.getAuthor(), actual.getAuthor());
+        EqualsBuilder.reflectionEquals(expected,actual,"id");
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Delete a book")
+    void delete_DeletesBookById_ExpectedSuccess() throws Exception {
+        Book book = new Book();
+                book.setId(4L);
+                book.setAuthor("John Doe");
+                book.setTitle("Book Title");
+                book.setDescription("Book Description");
+                book.setIsbn("9460306332081");
+                book.setPrice(BigDecimal.valueOf(10.99));
+        mockMvc.perform(
+                        delete("/books/{id}", 4L)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isNoContent());
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Update a book")
+    void updateBook_UpdatesBookDetails_ExpectedSuccess() throws Exception {
+        CreateBookRequestDto updateRequestDto = new CreateBookRequestDto()
+                .setAuthor("New Author")
+                .setTitle("Updated Title")
+                .setDescription("Updated Description")
+                .setIsbn("9460306332081")
+                .setPrice(BigDecimal.valueOf(15.99));
+
+        String jsonRequest = objectMapper.writeValueAsString(updateRequestDto);
+
+        mockMvc.perform(
+                        put("/books/{id}", 1L)
+                                .content(jsonRequest)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk());
```

Comment:
again, nobody knows after this test if book was updated or not. Only response status is checked what means nearly nothing